deps/ipc-util/ipc-util/pipe-windows.c

```diff
@@ -30,39 +38,45 @@ static inline void *create_full_access_security_descriptor()
 	if (!sd) {
 		return NULL;
 	}
+	//
+	//if (!InitializeSecurityDescriptor(sd, SECURITY_DESCRIPTOR_REVISION)) {
+	//	goto error;
+	//}
+	//
+	//if (!SetSecurityDescriptorDacl(sd, true, NULL, false)) {
+	//	goto error;
+	//}
 
-	if (!InitializeSecurityDescriptor(sd, SECURITY_DESCRIPTOR_REVISION)) {
-		goto error;
-	}
+	WCHAR *szSD = L"D:(D;OICI;GA;;;BG)(D;OICI;GA;;;AN)"
+		      L"(A;OICI;GRGWGX;;;AU)(A;OICI;GA;;;BA)";
 
-	if (!SetSecurityDescriptorDacl(sd, true, NULL, false)) {
-		goto error;
-	}
+	ConvertStringSecurityDescriptorToSecurityDescriptorW(
+		szSD, SDDL_REVISION_1, sd, NULL);
 
 	return sd;
 
-error:
-	free(sd);
-	return NULL;
+	//error:
+	//	free(sd);
+	//	return NULL;
```

Comment:
The commented out sections were accidentally left in, and that was part of the wrong commit. The reason for the changing the security descriptor creation means was because the MSDN docs and compiler warnings both tell you to not use a NULL security descriptor (which the old code did), because that gives complete control and ownership to everyone, whereas even an empty (but not NULL) descriptor would still limit some control and limit ownership to the process. In the end, I followed the example on MSDN for creating a DACL, incorporated that into the rest of the code, and verified with a test client that everything still worked as expected.